example/CMakeLists.txt

```
@@ -33,6 +33,8 @@ if(RANGES_BUILD_CALENDAR_EXAMPLE)
         message(STATUS "boost: ${Boost_LIBRARY_DIRS}")
         target_compile_definitions(calendar PRIVATE BOOST_NO_AUTO_PTR)
         target_compile_options(calendar PRIVATE -std=gnu++14)
-        set_target_properties(calendar PROPERTIES LINK_FLAGS "-L${Boost_LIBRARY_DIRS}")
+        if (${Boost_LIBRARY_DIRS})
+          set_target_properties(calendar PROPERTIES LINK_FLAGS "-L${Boost_LIBRARY_DIRS}")
```

```suggestion
        target_link_libraries(calendar PRIVATE Boost::date_time Boost::program_options)
```
Based on the discussion on the line below, this is how it should look like. Discard the other changes.

---

Not a master in this area, Can this type of change support older version of boost?

---

What actually matter is the CMake support for the Boost targets, which were [added in CMake 3.5](https://cmake.org/cmake/help/latest/release/3.5.html#modules). Range-v3 requires 3.6, so it's alright.

---

Updated this PR.